### PR TITLE
Extra device info

### LINKED ISSUE #81: We need function to check maximum size allowed as global and local range

### LINKED ISSUE #113: Dpctl needs to expose whether device supports 64 bit atomic operations

### PR DESCRIPTION
**Pull request by 1e-to**
The PR closes #113 and closes #81.

### CODE REVIEW

**Comment by PokhodenkoSA** [comment]
Linux CI fails.

File: dpctl/tests/test_sycl_device.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+##===----------------------------------------------------------------------===##
+
+import dpctl
+import unittest
+
+
+@unittest.skipIf(not dpctl.has_sycl_platforms(), "No SYCL platforms available")
+class TestSyclDevice(unittest.TestCase):
+    def test_get_max_compute_units(self):
+        q = dpctl.get_current_queue()
```
Comment (diptorupd) [inline dpctl/tests/test_sycl_device.py:32]: `get_current_queue` can throw. Maybe add a check?

**Review by diptorupd — COMMENTED** [review]
Looking good. Needs some minor clean ups and should be ready.

File: backends/tests/test_sycl_device_interface.cpp [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+
+struct TestDPPLSyclDeviceInterface : public ::testing::Test
+{
+    DPPLSyclDeviceRef OpenCL_cpu = nullptr;
+    DPPLSyclDeviceRef OpenCL_gpu = nullptr;
+    DPPLSyclDeviceRef OpenCL_level_zero = nullptr;
```
Comment (diptorupd) [inline backends/tests/test_sycl_device_interface.cpp:42]: Call it Level0_gpu. OpenCl and Level0 are different driver stacks for the same GPU device.

**Review by diptorupd — APPROVED** [review]
Looks very good. Please fix the one small naming change in C++ test case and it is good to go. Thanks!

**Comment by diptorupd** [comment]
@1e-to please rebase with master and I can merge it tomorrow.

**Pull request merged**